lib/jorb/backends/file.ex

```diff
@@ -0,0 +1,53 @@
+defmodule Jorb.Backends.File do
+  @moduledoc ~S"""
+  File-based backend for light queueing.
+
+  You probably don't want to use this for production load.
+  """
+  @behaviour Jorb.Backend
+  @queue_dir Path.join("tmp", "jorb")
+
+  def enqueue(queue_name, payload) do
+    ensure_queue_dir(queue_name)
+    message_name = (:erlang.system_time(:milli_seconds) |> Integer.to_string()) <> ".json"
+
+    # We're double-encoding the body here so we can have
+    # parity with how messages come out of SQS
+    final_payload = %{receipt_handle: message_name, body: Poison.encode!(payload)}
```

Comment:
I prefer a `with` pipeline, rather than using `!` functions in code that returns an `:ok`/`:error` tagged tuple.  If this function is allowed to raise, the typespec of `enqueue/2` would be `:ok | {:error, any} | no_return`.

Reply:
I think there's some general cruddery in both `Backend` implementations that I think could stand to benefit from a refactor to add a real boy `Message` type